### ISSUE TITLE
Deprecate cAdvisor flags in kubelet

### ISSUE THREAD

**Issue by timstclair** [kubernetes/kubernetes#19432]
Currently all cAdvisor flags are exposed through the cAdvisor binary. Many of the flags are irrelevant, or provide more controls than we want to give. Additionally, once we separate most cAdvisor functionality out of kubelet, most of thees flags will no longer make sense.

Open questions about the deprecation:
  - Which flags are existing customers using? We should probably send a survey to various kubernetes groups to find out.
  - Can we simply make the backwards incompatible change of removing the flags, or do we need to create "dummy" deprecated flags for each one in kubelet?

Prerequisite for https://github.com/kubernetes/kubernetes/issues/18770

@kubernetes/goog-node

**Comment by timstclair** [comment]
cAdvisor flag survey is up here: http://goo.gl/forms/NkAtiD4RXx

**Comment by yujuhong** [comment]
> cAdvisor flag survey is up here: http://goo.gl/forms/NkAtiD4RXx

Could we send this to a broader audience such as the mailing list?

**Comment by timstclair** [comment]
Yes, that's the plan. I'll send it out this afternoon.

**Comment by jdef** [comment]
is the plan to eventually run cadvisor as a peer process of the kubelet (as opposed to in-process)?

**Comment by vishh** [comment]
Kubelet will be self-sufficient. It will embed cAdvisor libraries to
collect the stats that are required for kubelet's functionality. A
standalone cAdvisor can be run on all nodes for monitoring purposes.

**Comment by timothysc** [comment]
I actually want to expose new params that are not currently exposed:  https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/cadvisor/cadvisor_linux.go#L48  until we actually have a daemonset cadvisor as the solution.  

Is the daemonset cadvisor a deliverable for 1.2?

/cc @kubernetes/rh-cluster-infra

**Comment by vishh** [comment]
> Is the daemonset cadvisor a deliverable for 1.2?

I don't think so. cc @timstclair

**Comment by timstclair** [comment]
Not for 1.2, hopefully for 1.3

**Comment by jdef** [comment]
does this mean that we'll be getting rid of the global cadvisor flags, replacing them with flag sets?

**Comment by timstclair** [comment]
Yes, that's the plan. There should be no changes to the standalone cAdvisor binary, but any project which is linking in cAdvisor will need to manually add (by calling package functions) any flags they want.

**Comment by jdef** [comment]
That's awesome news - thanks
